### ISSUE TITLE
receiving empty messages

### ISSUE THREAD

**Issue by sbourdeauducq** [davidgfnet/whatsapp-purple#356]
All messages sent to me in private conversations are received empty.

**Comment by vanfanel** [comment]
I'm getting the same: all I get are empty messages, no text to be read...

**Comment by pabloscasas** [comment]
In my case I only receive empty messages from a single contact in private conversations: in other private conversations I receive non empty messages. It happens since 24 hours ago with versions 0.8.6-42-g72611bf, 0.8.6-44-g6cde8b4 and 0.8.6-46-ge155c8d. With version 0.8.6-1-gf41d3c0 I get [Ciphered message received] in private conversations.

**Comment by davidgfnet** [comment]
@pabloscasas Any chance you guys can build a debug version (make cleanall && make debug) and you can send me a full trace with details (iOS, android, etc) over email? I can't fix it without a trace. I will add some more debugging messages later so we can get better traces.

**Comment by vanfanel** [comment]
@davidgfnet: of course, I'm used to GDB. But what do you mean by a "full trace"? Running it under gdb with no breakpoints, watches, etc.. won't yield anything special if the program doesn't segfault or something, and that's not the case.

**Comment by davidgfnet** [comment]
@vanfanel Compile it with "make debug" and it will print bucketloads of stuff to stdout, that's my trace :D

**Comment by pabloscasas** [comment]
The problem has disappeared with version 0.9.0-1-ga802c9a. Thanks a lot for fixing it !

**Comment by sbourdeauducq** [comment]
It works with most people for me, but there is still one person from whom I can only receive empty messages. Audio messages from this person work though.

**Comment by davidgfnet** [comment]
@sbourdeauducq Yeah audio messages are unencrypted. The same for viedeo messages. Can you receive msg from him in a group?

**Comment by andreas-42** [comment]
Also for me, the problem has unfortunately not disappeared with the new version a802c9a, although it happens with fewer contacts than in the previous version (I have found one contact where it still doesn't work, and one other for which the problem was fixed). As requested, I'll send you by e-mail a trace of a case that still doesn't work in the current version (messages from me to the other person work OK, received messages are empty). It is pretty much a fresh install of pidgin, with the WhatsApp account newly created, and default settings (resource Android-2.12.419).

**Comment by sbourdeauducq** [comment]
I tried creating a group and now I cannot connect at all anymore, getting "not-authorized" error.

**Comment by davidgfnet** [comment]
@andreas-42 That trace will be very interesting. Any idea of what platform is he/she using?

**Comment by andreas-42** [comment]
Yeah, by the way, the same here... I tried to create a new group a few days ago (don't remember exactly which whatsapp-purple version it was, but clearly a very recent build). It didn't work, and instead led to my WhatsApp account being permanently blocked immediately.

**Comment by davidgfnet** [comment]
@andreas-42 Uhm, I haven't really looked in group stuff so much in detail, they probably changed the way groups are managed, and we're still using the legacy method.

**Comment by sbourdeauducq** [comment]
Is there a way those whatsapp, inc. assholes will unblock my account or should I just get a new simcard? Thankfully those are cheap or easy to get in HK.

**Comment by davidgfnet** [comment]
@sbourdeauducq Send them an email, it worked for me twice but many years ago

**Comment by andreas-42** [comment]
@davidgfnet Maybe, I'll just let others create our groups now (and had to change the phone number that I use because of the blocking). The contact for which it doesn't work uses a Samsung Galaxy S3, Android 4.1.2. I'll send the trace now.

**Comment by davidgfnet** [comment]
@andreas-42 Would be great to knwo whatsapp version too. It is very very weird indeed! We had issues before with identity collision, who knows. Is he/she the owner (or original creator) of any group you talk to?

**Comment by andreas-42** [comment]
I've just sent you the trace. WhatsApp version of the contact is 3.12.367. The contact is not the owner of any group that I belong to.

**Comment by davidgfnet** [comment]
@andreas-42 Thanks! I think I just know what the issue is. Hes using an old version of whatsapp. Ill add the support back for it, but it he upgrades the problem will go away for sure! Thanks for the trace

**Comment by sbourdeauducq** [comment]
The answer to the email was that I had to contact them through the app, which I do not have. Also, I already got another simcard banned after I tried to register it through WART. Whatsapp, Inc. are a hateful bunch of fuckers.

**Comment by pabloscasas** [comment]
I was receiving empty messages from one of my contacts who used Whatsapp version 2.12.10 on Android. She upgraded to the latest Whatsapp version (2.12.429 now) and I immediately started to receive non-empty messages.

**Comment by davidgfnet** [comment]
@pabloscasas If someone is having this issue, please contact me so we can further debug that. I do know that people using old versions of the client send messages we cannot read, so it would be good to get a trace so I can fix it. thanks!

**Comment by pabloscasas** [comment]
davidgfnet, I'll let you know as soon as I find the problem again.

**Comment by Raaff** [comment]
I'm having this problem too, (with a contact that made pidgin crash, but worked after deleting his logs).

>Tree read:

>Tag: message
at[from]=xxxcrashxxx@s.whatsapp.net
at[id]=471F7XXXXXXXXXXXXXX
at[notify]=Contact Name
at[t]=1454582153
at[type]=text
Data:  ...
 Tag: enc
 at[type]=pkmsg
 at[v]=1
 Data: 3\010\0--\0*,\0,-\005\022!\005\/0, ...

>Received message stanza...
Axolotl exception (parseWhisperMessage): No such signedprekeyrecord! 26084 Unknown error
Tag: receipt
at[id]=471F7XXXXXXXXXXXXXX
at[t]=1454582153
at[to]=xxxcrashxxx@s.whatsapp.net
at[type]=retry
Data:  ...
 Tag: registration
 Data: >\0-/\/.-n ...
 Tag: retry
 at[count]=1
 at[id]=471F7XXXXXXXXXXXXXX
 at[t]=1454582153
 at[v]=1
 Data:  ...

**Comment by redstar3** [comment]
same here, I'm use wa4pidgin 0.9 on windows and still receiving empty messages. resources default from purple wa4pidgin 0.9

**Comment by Botoio** [comment]
Seems like you get empty messages just from the people who doesn't have your number in the contact list.

Thats how it is for me, am using 0.9 win version

**Comment by sbourdeauducq** [comment]
Third simcard, got it registered without the Whatsapp, Inc. scumbags banning it so far (as of now, registration works using yowsup). My contact upgraded Whatsapp on his phone, now I can receive messages correctly. @davidgfnet Thanks for whatsapp-purple! :+1:

**Comment by giordano** [comment]
> Seems like you get empty messages just from the people who doesn't have your number in the contact list.

This looks like a sensible guess, I think it's the same for me.

**Comment by redstar3** [comment]
>Seems like you get empty messages just from the people who doesn't have your number in the contact list. 
Thats how it is for me, am using 0.9 win version
@Botoio still get empty message, how can I fix this on my windows using 0.9 version. thanks

**Comment by menedas** [comment]
I get sometimes empty messages from same persons. But mostly it works. So its not a problem of the used WA version or if I'm in the contacts list.
For example yesterday evening and today morning a friend send me messages which where empty. Then when I tried to get debug messages it worked again. He said he doesn't changed anything.

**Comment by s09bQ5** [comment]
I have just received three empty group messages from three friends with messages from two other friends in that conversation being ok. @davidgfnet, do you still need debug logs? I have hex dumps of the buffers passed to read_tree.

**Comment by davidgfnet** [comment]
@s09bQ5 Logs always welcome! Go ahead send me an email :) Thanks!

**Comment by jaseg** [comment]
I just had that same problem for a new installation of whatsapp-pidgin with a account newly created using yowsup.

@davidgfnet [Here is the whatsapp-pidgin debug log](https://gist.github.com/jaseg/48ef2bb68e62532b2fb6). I suppose the interesting part is
```
Received message stanza...
Axolotl exception (parseWhisperMessage): No such signedprekeyrecord! 54177 Unknown error
Tag: receipt
```
The message's plaintext is ```Test```. I tried this to the same result with ```master``` as well as ```master``` with cherry-picked ```5e0d80a``` from devel as ```7621111``` would SIGABRT on account load and account creation.

**Comment by davidgfnet** [comment]
That abort might be related to some C++ exception?

**Comment by menedas** [comment]
That problem gets to be very annoying as the number of empty messages seems to be increasing. I often have to ask people to resend their messages. Is someone working on this?

I'm using the whatsapp4pidgin-0.9.0 for win. The problem exists since some months.

**Comment by caccamo88** [comment]
@davidgfnet 
have exactly same problem of @menedas that is receiving ONLY RANDOM empty messages (sending contact makes no difference) it's annoying because have to ask people to resend messages :( (basically 9 messages over 10 are ok, well maybe even more would say 95 over 100 are not empty)

**Comment by davidgfnet** [comment]
Just a reminder, this project is currently _unmantained_, this is, I abandoned its development. Feel free to pick it up if you want :)

**Comment by caccamo88** [comment]
:(((((((((((((((
